### PR TITLE
Enhance the SnippetsTask to support the new styles added for ESQL

### PR DESCRIPTION
**Pull request by ChrisHegarty**
This commit enhances the SnippetsTask's source matching to support the new styles added for ESQL [1].

[1] https://github.com/elastic/docs/pull/2697

### CODE REVIEW

**Comment by ChrisHegarty** [comment]
FYI - @abdonpijpelink

**Comment by ChrisHegarty** [comment]
@elasticsearchmachine rerun elasticsearch-ci/part-1

File: build-tools-internal/src/main/groovy/org/elasticsearch/gradle/internal/doc/SnippetsTask.groovy [hunk 1]
```diff
@@ -304,7 +304,7 @@ class SnippetsTask extends DefaultTask {
     }
 
     static Source matchSource(String line) {
-        def matcher = line =~ /\["?source"?,\s*"?([-\w]+)"?(,((?!id=).)*(id="?([-\w]+)"?)?(.*))?].*/
+        def matcher = line =~ /\["?source"?(?:\.[^,]+)?,\s*"?([-\w]+)"?(,((?!id=).)*(id="?([-\w]+)"?)?(.*))?].*/
```
Comment (breskeby) [inline build-tools-internal/src/main/groovy/org/elasticsearch/gradle/internal/doc/SnippetsTask.groovy:307]: I just trust your regex skills here

**Review by breskeby — APPROVED** [review]

File: build-tools-internal/src/main/groovy/org/elasticsearch/gradle/internal/doc/SnippetsTask.groovy (hunk 1 above)
Comment (nik9000) [reply build-tools-internal/src/main/groovy/org/elasticsearch/gradle/internal/doc/SnippetsTask.groovy:307]: It a new optional, non-capturing group that matching `.` followed by anything but a `,`.

**Review by nik9000 — APPROVED** [review]

**Pull request merged**